ConnectorRhino/ConnectorRhinoWebUI/Utils/RhinoIdleManager.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Collections.Generic;
+using Rhino;
+
+namespace ConnectorRhinoWebUI.Utils
+{
+  /// <summary>
+  /// Rhino Idle Manager is a helper util to manage deferred actions.
+  /// </summary>
+  public static class RhinoIdleManager
+  {
+    private static Dictionary<string, Action> calls = new Dictionary<string, Action>();
+
+    /// <summary>
+    /// Subscribe deferred action to RhinoIdle event to run it whenever Rhino become idle.
+    /// </summary>
+    /// <param name="action"> Action to call whenever Rhino become Idle.</param>
+    public static void SubscribeToIdle(Action action)
+    {
+      calls[action.Method.Name ?? Guid.NewGuid().ToString()] = action; 
+
+      RhinoApp.Idle += RhinoAppOnIdle;
```

Comment:
This can allow for multiple subscriptions if called in fast succession, and we might end up firing the idle event multiple times.